lib/ui/index.js

```diff
@@ -194,11 +193,13 @@ UI.prototype.prompt = function(questions, callback) {
 
   // If no callback was provided, automatically return a promise
   if (callback) {
-    inquirer.prompt(questions, callback);
-  } else {
-    return new Promise(function(resolve) {
-      inquirer.prompt(questions, resolve);
+    inquirer.prompt(questions).then(function(result) {
+      callback(result);
+    }).catch(function() {
+      callback(null);
```

Comment:
@Turbo87 @stefanpenner it seems with new `1.0.3` release, we should be able to re-throw ([commit link](https://github.com/SBoudrias/Inquirer.js/commit/19f5e1a4cadb1cc0c73f54c68691f9b61d241f0a)).

so then something like this should work:

```js
function rethrow(reason) {
  setTimeout(function() {
    throw reason;
  }, 0);
}
```